DIMS.nf

```diff
@@ -0,0 +1,133 @@
+    HMDBparts(params.hmdb_db_file, GenerateBreaks.out.breaks)
+
+    // Assign intensities to bins (breaks) per mzML file
+    AssignToBins(ConvertRawFile.out.combine(GenerateBreaks.out.breaks))
+
+    // Average intensities over technical replicates for each sample
+    AverageTechReplicates(AssignToBins.out.RData_files.collect(),
+                          AssignToBins.out.TIC_txt_files.collect(),
+                          MakeInit.out,
+                          params.nr_replicates, 
+                          analysis_id,
+                          matrix,
+                          GenerateBreaks.out.highest_mz)
+
+    // Peak finding per sample
+    PeakFinding(AverageTechReplicates.out.binned.collect().flatten().combine(GenerateBreaks.out.breaks))
+
+    // Spectrum peak finding per sample
+    SpectrumPeakFinding(PeakFinding.out.collect(), AverageTechReplicates.out.patterns)
+
+    // Peak grouping over samples: identified part
+    // PeakGrouping(HMDBparts.out.collect().flatten(), SpectrumPeakFinding.out, AverageTechReplicates.out.patterns)
+    PeakGrouping(HMDBparts.out.flatten(), SpectrumPeakFinding.out, AverageTechReplicates.out.patterns)
+
+    // Fill missing values in peak group list: identified part
+    FillMissing(PeakGrouping.out.grouped_identified, AverageTechReplicates.out.patterns)
+
+    // Collect filled peak group list: identified part
+    CollectFilled(FillMissing.out.collect(), AverageTechReplicates.out.patterns)
+
+    // Sum adducts of each metabolite per scan mode: identfied part
+    SumAdducts(CollectFilled.out.filled_pgrlist, 
+               AverageTechReplicates.out.patterns, 
+               HMDBparts_main.out.collect().flatten())
+
+    // Collect summed adducts parts
+    CollectSumAdducts(SumAdducts.out.collect())
+
+    // Generate final Excel file with Z-scores on adduct sums (pos + neg)
+    GenerateExcel(CollectSumAdducts.out.collect(), CollectFilled.out.filled_pgrlist.collect(), MakeInit.out, analysis_id, params.relevance_file)
+
+    // Generate violin plots 
+    GenerateViolinPlots(GenerateExcel.out.excel_file, analysis_id)
+
+    // Collect unidentified peaks
+    UnidentifiedCollectPeaks(SpectrumPeakFinding.out, PeakGrouping.out.peaks_used.collect())
+
+    // Peak grouping: unidentified part
+    UnidentifiedPeakGrouping(UnidentifiedCollectPeaks.out.collect(), AverageTechReplicates.out.patterns)
+
+    // Fill missing values in peak group list: unidentified part
+    UnidentifiedFillMissing(UnidentifiedPeakGrouping.out.grouped_unidentified, AverageTechReplicates.out.patterns)
+
+    // Calculate Z-scores for unidentified peak group list
+    UnidentifiedCalcZscores(UnidentifiedFillMissing.out, AverageTechReplicates.out.patterns)
+
+    // Create log files: Repository versions and Workflow params
+    //VersionLog(
+    //    Channel.of(
+    //        "${workflow.projectDir}/",
+    //        "${params.rawfiles_path}/",
+    //    ).collect()
+    //)
+    //Workflow_ExportParams()
```

Comment:
Deze ga je nog toevoegen?

Reply:
Toegevoegd met 
    VersionLog(
       Channel.of(
            "${workflow.projectDir}/"
            "${params.scripts_dir}/",
            "${params.hmdb_db_file}/",
            "${params.relevance_file}/",
        ).collect()
    )
    Workflow_ExportParams()
}